### PR TITLE
[FW][ADD] l10n_pe_pos: new module

### PR DESCRIPTION
**Pull request by rolandojduartem**
New point of sale for the Peruvian localization.

Forward-Port-Of: https://github.com/odoo/odoo/pull/143487

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by rolandojduartem** [comment]
Hi, @luisg123v, @desdelinux, @fvz-odoo and @adgu-odoo, please, could you review this PR?

 This PR is related to https://github.com/odoo/odoo/pull/151935. An inheritance would not be worthy.

**Comment by rhe-odoo** [comment]
Is it ok to accept the update of the Consumidor final customer? 
If not, we should maybe prevent that.

File: addons/l10n_pe_pos/static/src/overrides/components/partner_editor/partner_editor.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        this.changes.city_id = this.props.partner.city_id && this.props.partner.city_id.id;
+        this.changes.l10n_latam_identification_type_id =
+            this.props.partner.l10n_latam_identification_type_id &&
+            this.props.partner.l10n_latam_identification_type_id.id ||
+            this.pos.default_l10n_latam_identification_type_id;
+        this.changes.l10n_pe_district = this.props.partner.l10n_pe_district && this.props.partner.l10n_pe_district.id;
+        return res;
+    },
+    async confirm() {
+        const city = this.pos.models["res.city"].find((city) => city.id === parseInt(this.changes.city_id));
```
Comment (rhe-odoo) [inline addons/l10n_pe_pos/static/src/overrides/components/partner_editor/partner_editor.js:24]: If you're not in a pe country, you end up with a traceback. 
add: if (this.pos.isPeruvianCompany())

**Comment by rhe-odoo** [comment]
Does it make sense to invoice the Consumidor?
It doesn't relate to any real client and according to me it shouldn't be a valid practice.

**Comment by desdelinux** [comment]
> Does it make sense to invoice the Consumidor? It doesn't relate to any real client and according to me it shouldn't be a valid practice.

An invoice of Boleta document type is generated, it is a valid case in Peru.

Any transaction must be invoiced.

**Comment by desdelinux** [comment]
> Is it ok to accept the update of the Consumidor final customer?
> If not, we should maybe prevent that.

The name is not validated by SUNAT. And the businesses choose the one they want to give: "Consumidor Final", "Cliente Final", "Cliente genérico"

**Comment by rhe-odoo** [comment]
> > Is it ok to accept the update of the Consumidor final customer?
> > If not, we should maybe prevent that.
> 
> The name is not validated by SUNAT. And the businesses choose the one they want to give: "Consumidor Final", "Cliente Final", "Cliente genérico"

What about the different numbers? Right now, it's by default something like: 00000000
What if the default customer has something else? Is it still a legal stuff?

File: addons/l10n_pe_pos/static/src/overrides/components/partner_editor/partner_editor.js (hunk 1 above)
Comment (rolandojduartem) [reply addons/l10n_pe_pos/static/src/overrides/components/partner_editor/partner_editor.js:27]: good catch! Thanks, I applied your suggestion.

**Comment by rolandojduartem** [comment]
@rhe-odoo Technically, I applied the suggestions.

**Comment by desdelinux** [comment]
> What about the different numbers? Right now, it's by default something like: 00000000
> What if the default customer has something else? Is it still a legal stuff?

Yes, however, the most widely used is 00000000

**Comment by rhe-odoo** [comment]
It looks good to me.
Can you just update your commit message so it explain in a very simple way what this module do?
And don't forget to link the PR at the end of your commit message: see#odoo/151916

**Review by RicGR98 — COMMENTED** [review]
Thanks for the PR :sun_with_face: 

Just some small remarks. Regarding the camelCase, since this is a fw-port, I'm not sure it's worth changing it in this version since it will complexify all future fixes being forward ported, I'll let the final reviewer decide :)

File: addons/l10n_pe_pos/static/src/overrides/components/partner_editor/partner_editor.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        if (!this.pos.isPeruvianCompany()) {
+            return res;
+        }
+        this.intFields.push("city_id", "l10n_latam_identification_type_id", "l10n_pe_district");
+        this.changes.city_id = this.props.partner.city_id && this.props.partner.city_id.id;
+        this.changes.l10n_latam_identification_type_id =
+            this.props.partner.l10n_latam_identification_type_id &&
+            this.props.partner.l10n_latam_identification_type_id.id ||
+            this.pos.default_l10n_latam_identification_type_id;
+        this.changes.l10n_pe_district = this.props.partner.l10n_pe_district && this.props.partner.l10n_pe_district.id;
```
[the remark this answers is not in the document]
Comment (rolandojduartem) [reply addons/l10n_pe_pos/static/src/overrides/components/partner_editor/partner_editor.js:20]: No, the name in the backend is l10n_pe_district, we are keeping consistency at the moment to read from the frontend and backend.

[the remark this answers is not in the document]
Comment (rolandojduartem) [reply addons/l10n_pe_pos/static/src/overrides/components/partner_editor/partner_editor.js:20]: normally yes, but as I said, we are keeping consistency at the moment to read from the frontend and backend fields.

**Comment by rolandojduartem** [comment]
Hi, @RicGR98, I applied your suggestion or left relevant feedback in your comments, please, could you review it again?

File: addons/l10n_pe_pos/static/src/overrides/components/partner_editor/partner_editor.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (rolandojduartem) [reply addons/l10n_pe_pos/static/src/overrides/components/partner_editor/partner_editor.js:20]: A good example is https://github.com/odoo/odoo/blob/8e78196d7486143f1b144f9b6f5e85832e7b11e9/addons/point_of_sale/static/src/app/screens/partner_list/partner_editor/partner_editor.js#L47

File: addons/l10n_pe_pos/views/templates.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    <template id="ticket_validation_screen" inherit_id="point_of_sale.ticket_validation_screen">
+        <!-- We do not let the user create the invoice by themselves. That is why we hide the
+        "Get my Invoice" button -->
+        <xpath expr="//div[@id='get_my_invoice']" position="attributes">
+            <attribute name="t-if" add="(pos_order.company_id.country_code != 'PE')" separator=" and " />
+        </xpath>
+        <xpath expr="//div[@id='get_my_invoice']/.." position="after">
+            <h4 class="text-danger" t-if="pos_order.company_id.country_code == 'PE' and not pos_order.is_invoiced">
+                Invoice not available. You can contact us for more info.
```
[the remark this answers is not in the document]
Comment (desdelinux) [reply addons/l10n_pe_pos/views/templates.xml:12]: This message is the one that was accepted as valid in other modules.

Example: https://github.com/odoo/odoo/blob/17.0/addons/l10n_ar_pos/views/templates.xml#L11

[the remark this answers is not in the document]
Comment (rolandojduartem) [reply addons/l10n_pe_pos/views/templates.xml:12]: This comes from 17.0 https://github.com/odoo/odoo/blob/0a89ff8d764da1b551ce1d45d74255e7a3e5a603/addons/l10n_pe_pos/views/templates.xml#L12

**Comment by RicGR98** [comment]
Thanks for the changes/for the explanations.

I'll forward it for the final review :)

@smetl @oco-odoo if you want to take a look at the POS ?

@adgu-odoo as you approved the original PR

**Comment by fvz-odoo** [comment]
Hello @adgu-odoo and @jco-odoo 

This has been tested and it is working fine. All checks have passed and many reviews were also done here, we can go ahead and merge.

Thanks!

Fernando

**Comment by rhe-odoo** [comment]
@robodoo r+

**Pull request closed without merging**